lightning/src/ln/chan_utils.rs

```diff
@@ -1429,12 +1429,14 @@ impl<'a> TrustedCommitmentTransaction<'a> {
 
 		let htlc_redeemscript = get_htlc_redeemscript_with_explicit_keys(&this_htlc, self.opt_anchors(), &keys.broadcaster_htlc_key, &keys.countersignatory_htlc_key, &keys.revocation_key);
 
+		let sighashtype = if self.opt_anchors() { SigHashType::SinglePlusAnyoneCanPay } else { SigHashType::All };
```

Comment:
I believe this is also used to add the second signature, no? `counterparty_signature` may also be SIGHASH_SINGLE, this would be the second (non-SINGLE) signature.

Reply:
The type for the first signature (`remotehtlcsig` / `counterparty_signature`) is set to `SinglePlusAnyoneCanPay` when `opt_anchors()` is true.

The type for the second signature(`localhtlcsig` / `signature`) is always `SigHashType::All`:

```
		htlc_tx.input[0].witness.push(counterparty_signature.serialize_der().to_vec());
		htlc_tx.input[0].witness.push(signature.serialize_der().to_vec());
		htlc_tx.input[0].witness[1].push(sighashtype as u8);
		htlc_tx.input[0].witness[2].push(SigHashType::All as u8);
```